app/store.go

```diff
@@ -291,29 +292,17 @@ func (s *StoreApp) Commit() (res abci.ResponseCommit) {
 // Note: in tendermint 0.17, the genesis file is passed
 // in here, we should use this to trigger reading the genesis now
 func (s *StoreApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitChain) {
-	bz, err := ioutil.ReadFile(s.genesisFile)
-	if err != nil {
-		// I'm sorry Alex, but there is no other way :(
-		// https://github.com/tendermint/abci/issues/165#issuecomment-353704015
-		// "Regarding errors in general, for messages that don't take
-		//  user input like Flush, Info, InitChain, BeginBlock, EndBlock,
-		// and Commit.... There is no way to handle these errors gracefully,
-		// so we might as well panic."
-		panic(err)
-	}
-
-	return s.InitChainWithGenesis(req, bz)
+	return s.InitChainWithAppState(req)
 }
 
-// InitChainWithGenesis mocks out what we want for tendermint 0.17
-func (s *StoreApp) InitChainWithGenesis(req abci.RequestInitChain,
-	data []byte) abci.ResponseInitChain {
-
-	err := s.parseGenesis(data, s.initializer)
+// InitChainWithAppState mocks out what we want for tendermint 0.17
```

Comment:
You can collapse this into InitChain.
They are now the same functionality, let's just make them one.